Fix flaky sendermonitor and gaspricemonitor tests

**Is your feature request related to a problem? Please describe.**
A clear and concise description of what the problem is. Ex. I'm always frustrated when [...]

Certain tests in `pm/sendermonitor_test.go` and `eth/gaspricemonitor_test.go` are flaky and fail in CI sometimes.

`pm/sendermonitor_test.go` example:

```
--- FAIL: TestQueueTicketAndSignalMaxFloat (0.05s)
    sendermonitor_test.go:258:
            Error Trace:    sendermonitor_test.go:258
            Error:          Not equal:
                            expected: 0x3
                            actual  : 0x2
            Test:           TestQueueTicketAndSignalMaxFloat
    sendermonitor_test.go:259:
            Error Trace:    sendermonitor_test.go:259
            Error:          Not equal:
                            expected: 0x2
                            actual  : 0x3
            Test:           TestQueueTicketAndSignalMaxFloat
```

`eth/gaspricemonitor_test.go` example:

```
--- FAIL: TestStart_Polling (4.00s)
    gaspricemonitor_test.go:149: 
        	Error Trace:	gaspricemonitor_test.go:149
        	Error:      	Not equal: 
        	            	expected: 555
        	            	actual  : 777
        	            	
        	            	Diff:
        	            	--- Expected
        	            	+++ Actual
        	            	@@ -1,2 +1,2 @@
        	            	-(*big.Int)(555)
        	            	+(*big.Int)(777)
        	            	 
        	Test:       	TestStart_Polling
```

**Describe the solution you'd like**
A clear and concise description of what you want to happen.

We should investigate the reason (likely related to timing with `time.Ticker` and concurrent processing) for this flaky test behavior and fix the issue.

--- a/eth/gaspricemonitor_test.go
+++ b/eth/gaspricemonitor_test.go
@@ -95,7 +95,7 @@ func TestStart_Polling(t *testing.T) {
 	gasPrice3 := big.NewInt(888)
 	gpo := newStubGasPriceOracle(gasPrice1)
 
-	pollingInterval := 1 * time.Second
+	pollingInterval := 1 * time.Millisecond
 	gpm := NewGasPriceMonitor(gpo, pollingInterval)
 
 	assert := assert.New(t)
@@ -139,11 +139,10 @@ func TestStart_Polling(t *testing.T) {
 	// sync sleep finishes, the monitor
 	// should have decreased its own gas price
 	go func() {
-		time.Sleep(1 * pollingInterval)
 		gpo.SetGasPrice(gasPrice2)
 	}()
 
-	time.Sleep(2 * pollingInterval)
+	time.Sleep(100 * time.Millisecond)
 
 	assert.Greater(gpo.Queries(), 0)
 	assert.Equal(gasPrice2, gpm.GasPrice())
@@ -154,11 +153,10 @@ func TestStart_Polling(t *testing.T) {
 	// sync sleep finishes, the monitor
 	// should have increased its own gas price
 	go func() {
-		time.Sleep(1 * pollingInterval)
 		gpo.SetGasPrice(gasPrice3)
 	}()
 
-	time.Sleep(2 * pollingInterval)
+	time.Sleep(100 * time.Millisecond)
 
 	// There should be more queries now
 	assert.Greater(gpo.Queries(), queries)
@@ -184,7 +182,7 @@ func TestStart_Polling_ContextCancel(t *testing.T) {
 	// Cancel polling loop
 	cancel()
 
-	time.Sleep(1 * time.Second)
+	time.Sleep(100 * time.Millisecond)
 
 	// Ensure there are no more queries
 	assert.Equal(t, gpo.queries, queries)
@@ -215,7 +213,7 @@ func TestStop(t *testing.T) {
 	err = gpm.Stop()
 	assert.Nil(err)
 
-	time.Sleep(1 * time.Second)
+	time.Sleep(100 * time.Millisecond)
 
 	// Ensure there are no more queries
 	assert.Equal(queries, gpo.queries)

--- a/pm/sendermonitor_test.go
+++ b/pm/sendermonitor_test.go
@@ -231,12 +231,15 @@ func TestQueueTicketAndSignalMaxFloat(t *testing.T) {
 	qc = &queueConsumer{}
 	go qc.Wait(2, sm)
 
+	time.Sleep(10 * time.Millisecond)
+
 	err = sm.AddFloat(addr2, big.NewInt(5))
 	require.Nil(err)
 	err = sm.AddFloat(addr, big.NewInt(5))
 	require.Nil(err)
 
-	time.Sleep(time.Millisecond * 20)
+	time.Sleep(10 * time.Millisecond)
+
 	// Order of tickets should reflect order that AddFloat()
 	// was called
 	tickets = qc.Redeemable()

